gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/kmeans/KMeansTest.java

```diff
@@ -0,0 +1,173 @@
+
+    assertTrue(expectedNearestCenter.size() == nearestCenter.size() &&
+      expectedNearestCenter.containsAll(nearestCenter));
+  }
+
+  @Test
+  public void testCountAppender() throws Exception {
+    List<Tuple3<Integer, Point, Long>> nearestCenterWithAppendedCounter =
+      points.map(new SelectNearestCenter()).withBroadcastSet(centroids, "centroids")
+        .map(new CountAppender())
+        .collect();
+
+    assertTrue(expectedNearestCenterWithAppendedCounter.size() ==
+      nearestCenterWithAppendedCounter.size() &&
+      expectedNearestCenterWithAppendedCounter.containsAll(nearestCenterWithAppendedCounter));
+  }
+
+  @Test
+  public void testCentroidAccumulator() throws Exception {
+    List<Tuple3<Integer, Point, Long>> centroidsWithAccumulatedPoints =
+      points.map(new SelectNearestCenter()).withBroadcastSet(centroids, "centroids")
+        .map(new CountAppender())
+        .groupBy(0).reduce(new CentroidAccumulator()).collect();
+
+    assertTrue(expectedCentroidsWithAccumulatedPoints.size() ==
+      centroidsWithAccumulatedPoints.size() &&
+      expectedCentroidsWithAccumulatedPoints.containsAll(centroidsWithAccumulatedPoints));
+  }
+
+  @Test
+  public void testCentroidAverager() throws Exception {
+    List<Centroid> averageCentroids =
+      points.map(new SelectNearestCenter()).withBroadcastSet(centroids, "centroids")
+        .map(new CountAppender())
+        .groupBy(0).reduce(new CentroidAccumulator()).map(new CentroidAverager()).collect();
+
+    assertTrue(expectedAverageCentroids.size() == averageCentroids.size() &&
+      expectedAverageCentroids.containsAll(averageCentroids));
+  }
+
+
+  //Integration Test
+  @Test
+  public void testKMeansAlgorithm() throws Exception {
+    LogicalGraph logicalGraph = getLoaderFromString(inputGdlGraph).getLogicalGraphByVariable("g");
+    LogicalGraph output =
+      (LogicalGraph) new KMeans(20, 3, "lat", "long")
+        .execute(logicalGraph);
+    DataSet<V> vertices = (DataSet<V>) output.getVertices();
+    DataSet<V> comparisonVertices = vertices.filter(
+      vertex -> vertex.hasProperty("cluster_lat") && vertex.hasProperty("cluster_long") &&
+        vertex.hasProperty("cluster_id") && vertex.hasProperty("long") &&
+        vertex.hasProperty("lat"));
+
+    assertTrue(comparisonVertices.collect().size() == vertices.collect().size());
+
+    //Mapping function to see the values of the vertex properties
+    DataSet<String> properties = vertices.map(
+      (MapFunction<V, String>) vertex -> vertex.getPropertyValue("cluster_lat").toString() + " " +
+        vertex.getPropertyValue("cluster_long").toString() + " " +
+        vertex.getPropertyValue("cluster_id").toString() + " " +
+        vertex.getPropertyValue("long").toString() + " " +
+        vertex.getPropertyValue("lat").toString());
+
+    for (String s : properties.collect()) {
+      System.out.println(s);
```

Comment:
Is this a relict? No printing in test classes

Reply:
Was a relict. Removed from test class